client/session.go

```diff
@@ -153,18 +153,27 @@ func (s *session) Open() error {
 	s.seriesIteratorsPool.Init()
 	s.Unlock()
 
-	currTopologyMap := s.topo.GetAndSubscribe(s.topoMapCh)
+	currTopologyMap, topologyWatch, err := s.topo.GetAndSubscribe()
+	if err != nil {
+		return err
```

Comment:
I moved the GetAndSubscribe() and error check up before setting s.state to open, the s was locked so no two consumers can call it at the same time, no?